orchagent/dash/dashrouteorch.cpp

```diff
@@ -0,0 +1,994 @@
+#include <cassert>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <exception>
+#include <inttypes.h>
+#include <algorithm>
+
+#include "converter.h"
+#include "dashrouteorch.h"
+#include "macaddress.h"
+#include "orch.h"
+#include "sai.h"
+#include "saiextensions.h"
+#include "swssnet.h"
+#include "tokenize.h"
+
+using namespace std;
+using namespace swss;
+
+extern std::unordered_map<std::string, sai_object_id_t> gVnetNameToId;
+extern sai_dash_eni_api_t* sai_dash_eni_api;
+extern sai_dash_outbound_routing_api_t* sai_dash_outbound_routing_api;
+extern sai_dash_inbound_routing_api_t* sai_dash_inbound_routing_api;
+extern sai_object_id_t gSwitchId;
+extern size_t gMaxBulkSize;
+
+static std::unordered_map<std::string, sai_outbound_routing_entry_action_t> sOutboundAction =
+{
+    { "vnet", SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET },
+    { "vnet_direct", SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET_DIRECT },
+    { "route_direct", SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_DIRECT },
+    { "drop", SAI_OUTBOUND_ROUTING_ENTRY_ACTION_DROP }
+};
+
+DashRouteOrch::DashRouteOrch(DBConnector *db, vector<string> &tableName) :
+    outbound_routing_bulker_(sai_dash_outbound_routing_api, gMaxBulkSize),
+    inbound_routing_bulker_(sai_dash_inbound_routing_api, gMaxBulkSize),
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+}
+
+bool DashRouteOrch::addEniEntry(const string& eni)
+{
+    SWSS_LOG_ENTER();
+
+    if (eni_entries_[eni].eniEntryAdded)
+    {
+        SWSS_LOG_ERROR("ENI entry already exists for %s", eni.c_str());
```

Comment:
Lets change such 'already exists' or 'object not found' cases to WARNING as northbound APIs are expected to be idempotent

Reply:
Will do.